tuna/connection.py

```diff
@@ -174,14 +257,18 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       if abort is not None and chk_abort_file(self.id, self.logger):
         self.logger.warning('Machine %s aborted command execution: %s', self.id,
                             cmd)
-        return None, None, None
+        return i_var, o_var, e_var
 
     self.logger.error('cmd_exec retries exhausted, giving up')
-    return None, None, None
+    return i_var, o_var, e_var
 
-  def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+  def exec_command(self, cmd: str, timeout: int=int(SSH_TIMEOUT), abort: Optional[bool]=None,\
+  proc_line: Union[Any, None]= None) -> Tuple[Union[int, Any], Union[TextIO, StringIO, \
+  ChannelFile],Union[ChannelStderrFile, StringIO]]:
     # pylint: disable=too-many-nested-blocks, too-many-branches
     """Function to exec commands"""
+    o_var: ChannelFile
+    e_var: ChannelStderrFile
     _, o_var, e_var = self.exec_command_unparsed(cmd, timeout, abort)
     try:
       if not o_var:
```

Comment:
Following return type will be a result of POpen in exec_command_unparsed, which would be StringIO.

Reply:
It is returning both StringIO and Channel file objects(input/output/error types) so the return types have to have same type hints. 
Line number : 232 retruns IO objects.
Line number 244: retruns ChannelObjects

Reply:
254 , out channel is paramiko channel type and ovar.channel doesnot belog to StringIO object type so changing o_var to StringIO doesnot yelied compatible type, line 254 assignment throws error.

Reply:
Disregarding the implications of the type hinting, for reference, this is the real type of the returned values at this point.